common/extent-tree-utils.c

```diff
@@ -285,3 +285,93 @@ int btrfs_record_file_extent(struct btrfs_trans_handle *trans,
 	}
 	return ret;
 }
+
+int btrfs_record_file_extent_comp(struct btrfs_trans_handle *trans,
```

Comment:
Please do no use `comp` as a short name for `compress`, it's a little confusing.

But it will no longer be a problem if we address the next comment.

Reply:
It turns out that we should not use `btrfs_record_file_extent()` except for convert.

The extra handling inside `btrfs_record_file_extent()` is specially designed for convert, where we may have an existing data extent from the convert image, thus it is not a good way to support compressed data extents.

For mkfs cases, I'll craft a dedicated help for it with a more expandable interface during my cleanup.